### PR TITLE
自定义log输出

### PR DESCRIPTION
**Pull request by sipt**
1.添加 const DefaultLoggerFuncCallDepth=2
2.开放 logger输出方法
3.添加用户自定义输出级别LeveCustom
4.提取重得代码formatLogTime

### CODE REVIEW

File: logs/log.go [hunk 1]
```diff
@@ -199,6 +201,15 @@ func (bl *BeeLogger) writeMsg(logLevel int, msg string) error {
...
 		msg = "[" + filename + ":" + strconv.FormatInt(int64(line), 10) + "]" + msg
 	}
+	bl.OutputMsg(msg, logLevel)
+}
+
+//open output log for custom
+func (bl *BeeLogger) OutputMsg(msg string, logLevel int) {
+	when := time.Now()
+	if LevelCustom != logLevel {
+		msg = formatLogTime(when) + msg
```
Comment (JessonChan) [inline logs/log.go]: msg can't append a time header,check the code in logs/es/es.go L57~60

File: logs/log.go [hunk 2]
```diff
@@ -45,7 +45,8 @@ import (
 
 // RFC5424 log message levels.
 const (
-	LevelEmergency = iota
+	LevelCustom = iota //level for custom
```
Comment (JessonChan) [inline logs/log.go]: what happens when the log level set to a LevelCustom?

Comment (sipt) [inline logs/log.go]: Users can customize the level of log;
For example,
the level of business log: [INFO][DEBUG][ERROR]....  
the level of database log:[SQL] [PARAMS] [RESULT] [INSERT] [SELECT] [UPDATE]...
their can be different.

File: logs/log.go (hunk 1 above)
Comment (sipt) [inline logs/log.go]: missed it....

Comment (sipt) [inline logs/log.go]: missed it.
Custom time format is a problem

File: logs/log.go (hunk 2 above)
Comment (JessonChan) [inline logs/log.go]: It's a good idea,but the code is not so elegant.

**Comment by sipt** [comment]
@JessonChan 
I have restored "formatLogTime".

**Pull request closed without merging**